content/docs/1.1.1/advanced-resources/rancher-cluster-restore.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Restore cluster with a Rancher snapshot
+weight: 4
+---
+
+This doc describes what users need to do after restore the cluster with a Rancher snapshot.
+
+## Assumptions:
+- Most of the data and the underlying disks still exist in the cluster before the restore and can be directly reused then.
+- There is a backupstore holding all volume data.
+- The setting [`Disable Revision Counter`](../../references/settings/#disable-revision-counter) is false. (It's false by default.) Otherwise, users need to manually check if the data among volume replicas are consistent, or directly restore volumes from backup.
+
+## Expectation:
+- All settings and node & disk configs will be restored.
+- As long as the valid data still exists, the volumes can be recovered without using backup. In other words, we will try to avoid restoring backups, which may help reduce Recovery Time Objective (RTO) as well as save bandwidth:
+- Detect the invalid or out-of-sync replicas as long as the related volume still contains a valid replica after the restore.
+
+## Behaviors & Requirement of Rancher restore
+- According to [the Rancher restore article](https://rancher.com/blog/2018/2018-05-30-recover-rancher-kubernetes-cluster-from-backup/), you have to restart the Kubernetes components on all nodes. Otherwise, there will be tons of resource update conflicts in Longhorn.
+
+## Actions after the restore
+- Restart all Kubernetes components for all nodes. See the above link for more details.
+
+- Kill all longhorn manager pods then Kubernetes will automatically restart them. Wait for conflicts in longhorn manager pods to disappear.
+
+- All volumes may be reattached. If a Longhorn volume is used by a single pod, users need to shut down then recreate it. For Deployments or Statefulsets, Longhorn will automatically kill then restart the related pods.
+    ```sh
+    docker restart kube-apiserver kubelet kube-controller-manager kube-scheduler kube-proxy
+    docker ps | grep flannel | cut -f 1 -d " " | xargs docker restart
+    docker ps | grep calico | cut -f 1 -d " " | xargs docker restart
```

Comment:
This should be placed in the second statement (line: 24), but I would suggest no need to add any snippet because users can be guided to the external documentation as a reference.

Reply:
Agree. Personally, I am still using `docker` but some users already switch to `crictl`. I will remove this snippet. we can assume users can handle the restart by themselves or refer to the external doc we provide. @jenting Is this good for you?

Reply:
I vote to refer external doc.